src/main/java/duke/command/AllCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import java.util.Arrays;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import javafx.util.Pair;
+
+/**
+ * All Command for Duke application.
+ *
+ * @author Farrel Dwireswara Salim
+ */
+public class AllCommand implements Command {
+    private final String[] tags;
+
+    /**
+     * Constructs a new AllCommand instance based on tags.
+     *
+     * @param chosenTags the chosen tags.
+     * @throws DukeException If the chosen tags are empty.
+     */
+    public AllCommand(String chosenTags) throws DukeException {
+        if (chosenTags == null || chosenTags == "") {
+            throw new DukeException("The tags after any command "
+                    + "must be a non-empty string!");
+        }
+
+        // Filter out all empty tags
+        this.tags = Arrays.stream(chosenTags.split(","))
+                .map(String::trim)
+                .filter(text -> !text.isEmpty())
```

Comment:
Use of double negative here, might be good to introduce isNotEmpty() [to remove double negatives](https://refactoring.com/catalog/removeDoubleNegative.html)